src/test/bridges/notional/NotionalUnit.t.sol

```diff
@@ -0,0 +1,490 @@
+        _x = bound(_x, 1e6, 1e10);
+        Info memory info;
+        info.currencyId = USDC_CURRENCYID;
+        info.maturity = uint40(notionalView.getActiveMarkets(info.currencyId)[0].maturity);
+        info.interactionNonce = 1;
+        address fcashToken = FCASH_FACTORY.deployWrapper(info.currencyId, info.maturity);
+        _lend(_x, USDC, info);
+        uint256 balance = ERC20(fcashToken).balanceOf(address(ROLLUP_PROCESSOR));
+        if (balance == 0) {
+            revert("receive fcash for lending");
+        }
+    }
+
+    function testCUSDCLendWithdrawCUSDC(uint256 _x, bool _overMaturity) public {
+        _x = bound(_x, 1e9, 1e12);
+        Info memory info;
+        info.currencyId = USDC_CURRENCYID;
+        info.maturity = uint40(notionalView.getActiveMarkets(info.currencyId)[0].maturity);
+        info.interactionNonce = 1;
+        address fcashToken = FCASH_FACTORY.deployWrapper(info.currencyId, info.maturity);
+        _lend(_x, CUSDC, info);
+        uint256 prevBalance = ERC20(CUSDC).balanceOf(address(ROLLUP_PROCESSOR));
+        uint256 withdrawAmount = ERC20(fcashToken).balanceOf(address(ROLLUP_PROCESSOR));
+        info.underlying = false;
+        info.interactionNonce = 2;
+        if (_overMaturity) {
+            vm.warp(block.timestamp + 90 days);
+        }
+        _withdraw(withdrawAmount, USDC, CUSDC, info);
+        uint256 redeemedBalance = ERC20(CUSDC).balanceOf(address(ROLLUP_PROCESSOR)) - prevBalance;
+        if ((redeemedBalance * 10000) / _x < 9900) {
+            revert("should take most of money back");
+        }
+        if (ERC20(fcashToken).balanceOf(address(ROLLUP_PROCESSOR)) != 0) {
+            revert("fcash should be burned");
+        }
+    }
+
+    function testUSDCWithdraw(uint256 _x, bool _overMaturity) public {
+        _x = bound(_x, 1e6, 1e10);
+        Info memory info;
+        info.currencyId = USDC_CURRENCYID;
+        info.interactionNonce = 1;
+        info.maturity = uint40(notionalView.getActiveMarkets(info.currencyId)[0].maturity);
+        address fcashToken = FCASH_FACTORY.deployWrapper(info.currencyId, info.maturity);
+        _lend(_x, USDC, info);
+        uint256 prevBalance = ERC20(USDC).balanceOf(address(ROLLUP_PROCESSOR));
+        uint256 withdrawAmount = ERC20(fcashToken).balanceOf(address(ROLLUP_PROCESSOR));
+        info.underlying = true;
+        info.interactionNonce = 2;
+        if (_overMaturity) {
+            vm.warp(block.timestamp + 90 days);
+        }
+        _withdraw(withdrawAmount, USDC, CUSDC, info);
+        uint256 redeemedBalance = ERC20(USDC).balanceOf(address(ROLLUP_PROCESSOR)) - prevBalance;
+        if ((redeemedBalance * 10000) / _x < 9900) {
+            revert("should take most of money back");
+        }
+        if (ERC20(fcashToken).balanceOf(address(ROLLUP_PROCESSOR)) != 0) {
+            revert("fcash should be burned");
+        }
+    }
+
+    function testWBTCLend(uint256 _x) public {
```

Comment:
Please generalize these tests so that they fetch tokens form Notional view contract and iteratively runs these tests in a for loop.